### PR TITLE
[BPK-3762] Flare component

### PR DESCRIPTION
**Pull request by shaundon**
Skeleton component, doesn't do anything useful yet. Because of our new monopackage, if we merge and release this it'll be available to consumers. Should we:

1. Mark it as beta in some way?
1. Don't worry, it won't be documented so people are unlikely to find it.
1. Not merge this PR until the native side is done on at least one platform?

### CODE REVIEW

File: lib/bpk-component-flare/src/BpkFlare-test.android.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import commonTests from './BpkFlare-test.common';
+
+jest.mock('react-native', () => {
```
Comment (tiagoengel) [inline lib/bpk-component-flare/src/BpkFlare-test.android.js]: Is this necessary?

File: lib/bpk-component-flare/src/BpkFlare.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+const BpkFlare = (props: Props) => {
+  const { children, insetPaddingMode, style: userStyle, ...rest } = props;
+
+  const style = [styles.base];
+  if (userStyle) {
+    style.push(userStyle);
+  }
+
+  return (
+    <View style={style} {...rest}>
```
Comment (tiagoengel) [inline lib/bpk-component-flare/src/BpkFlare.js]: Can we replace the view here with a `NativeFlare` component? The files would be `NativeFlare.ios` and `NativeFlare.android`  and they can just return a view and spread the props for now,

File: lib/bpk-component-flare/src/BpkFlare-test.android.js (hunk 1 above)
Comment (shaundon) [reply lib/bpk-component-flare/src/BpkFlare-test.android.js]: That came from the `create-component` script 🤷‍♂

File: lib/bpk-component-flare/src/BpkFlare.js (hunk 2 above)
Comment (shaundon) [reply lib/bpk-component-flare/src/BpkFlare.js]: Sure, will do that

**Comment by tiagoengel** [comment]
About marking as beta, I wouldn't worry, people will not use if is not documented/doesn't do anything

File: lib/bpk-component-flare/src/BpkFlare-test.android.js (hunk 1 above)
Comment (tiagoengel) [reply lib/bpk-component-flare/src/BpkFlare-test.android.js]: Ahh, so it's not necessary then :)

Comment (shaundon) [reply lib/bpk-component-flare/src/BpkFlare-test.android.js]: I'll shut it down!

File: lib/bpk-component-flare/src/NativeFlare.android.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import { View } from 'react-native';
+
+const NativeFlare = ({ ...rest }) => <View {...rest} />;
```
Comment (tiagoengel) [inline lib/bpk-component-flare/src/NativeFlare.android.js]: nit: Can just do `NativeFlare = View` and would make flow stop complaining as well

**Pull request merged**